File: ejemplos/myapiexample/app/src/main/java/com/example/myapiexample/api/PokemonApiService.java
package com.example.myapiexample.api;

import com.example.myapiexample.api.response.PokemonResponse;

import retrofit2.Call;
import retrofit2.http.GET;

public interface PokemonApiService {

    @GET("pokemon")
    Call<PokemonResponse> getPokemon();

}

File: ejemplos/PokemonDesk/app/src/main/java/cl/inacap/apps/pokemondesk/api/PokemonApiService.kt
package cl.inacap.apps.pokemondesk.api

import cl.inacap.apps.pokemondesk.api.response.PokemonListResponse
import cl.inacap.apps.pokemondesk.api.response.PokemonResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface PokemonApiService {
    @GET("pokemon")
    fun getPokemonList(@Query("offset") offset: Int = 0,
                       @Query("limit") limit: Int = 20): Call<PokemonListResponse>

    @GET("pokemon/{name}")
    fun getPokemon(@Path("name") name:String): Call<PokemonResponse>
}